raiden_contracts/data/source/raiden/TokenNetwork.sol

```diff
@@ -316,6 +316,39 @@ contract TokenNetwork is Utils {
         return channel_identifier;
     }
 
+    /// @notice Opens a new channel between `participant1` and `participant2`
+    /// and deposits for `participant1`. Can be called by anyone
+    /// @param participant1 Ethereum address of a channel participant
+    /// @param participant2 Ethereum address of the other channel participant
+    /// @param settle_timeout Number of blocks that need to be mined between a
+    /// call to closeChannel and settleChannel
+    /// @param participant1_total_deposit The total amount of tokens that
+    /// `participant1` will have as deposit
+    function openChannelWithDeposit(
+        address participant1,
+        address participant2,
+        uint256 settle_timeout,
+        uint256 participant1_total_deposit
+    )
+        public
+        isSafe
+        settleTimeoutValid(settle_timeout)
+        returns (uint256)
+    {
+        uint256 channel_identifier;
```

Comment:
Should we add a require for `participant1_total_deposit > 0` here? Otherwise the call to `participant1_total_deposit` will fail.

Reply:
It will already fail with a descriptive error, AFAICS, so I don't see how another require would improve the situation.

Reply:
True, the only advantage would be failing earlier, but it should fail during gas estimation already.